src/utils.rs

```diff
@@ -0,0 +1,31 @@
+use anyhow::Context;
+use reqwest::blocking::Response;
+use serde::de::DeserializeOwned;
+
+pub trait ResponseExt {
+    fn custom_error_for_status(self) -> anyhow::Result<Response>;
+    fn json_annotated<T: DeserializeOwned>(self) -> anyhow::Result<T>;
+}
+
+impl ResponseExt for Response {
+    fn custom_error_for_status(self) -> anyhow::Result<Response> {
+        match self.error_for_status_ref() {
+            Ok(_) => Ok(self),
+            Err(err) => {
+                let body = self.text()?;
+                Err(err).context(format!("Body: {:?}", body))
+            }
+        }
+    }
+
+    fn json_annotated<T: DeserializeOwned>(self) -> anyhow::Result<T> {
+        let text = self.text()?;
+        serde_json::from_str::<T>(&text).with_context(|| {
+            format!(
+                "Cannot deserialize type `{}` from\n{}",
+                std::any::type_name::<T>(),
+                text
```

Comment:
Maybe we could try to deserialize to `serde_json::Value` first so that if it's valid JSON we'll at least know and be able to format a bit better than plain text.

Reply:
The error message isn't really much better if we parse from `serde_json::Value`, since `serde` by default doesn't include the "path" to the error in the error message.

Reply:
Yea but the formatting of `serde_json::Value` is nicer than a plain string which can be hard to parse by eye.

Reply:
That's true, although when you get a message like "missing field at column 1000", after formatting to `serde_json::Value`, there won't be any column 1000 in the debug output anymore :) But it's probably more useful anyway to format the response as JSON than to hunt column numbers on a very long line of text.